### PR TITLE
DS-4133 Support Ruby 3.3

### PR DESCRIPTION
**Pull request by ildomar-grings**
## Overview

[DS-4133](https://loyaltynz.atlassian.net/browse/DS-4133)

## What is the change?
- Upgrade Ruby to version 3.3.0
- Upgrade Bundler to version 2.3.10
- Upgrade the `yard` version to fix a vulnerability
- Add version checker to the CI

## Notes

[DS-4133]: https://loyaltynz.atlassian.net/browse/DS-4133?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Change log
 
+## v1.5.0
+
+- Support Ruby 3.3
```
Comment (davidoram) [inline CHANGELOG.md:5]:
```suggestion
- Support Ruby 3.3 [DS-4133](https://loyaltynz.atlassian.net/browse/DS-4133)
```

**Review by davidoram — APPROVED** [review]
Approved with minor CHANGELOG.md addition

**Review by nickpiercy95 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (pond) [reply CHANGELOG.md:5]: Yep, definitely _do not_ leak your internal JIRA ticket URLs in public gems!

File: Rakefile [hunk 2]
```diff
@@ -13,4 +13,14 @@ begin
...
+end
+
+desc 'Check if latest version in CHANGELOG.md matches with current version number'
+task :check_version do
+  changelog = File.join(File.dirname(__FILE__), 'CHANGELOG.md')
+  raise "missing CHANGELOG.md" unless File.exist?(changelog)
+
+  if File.read(changelog).match(/[0-9]+\.[0-9]+\.[0-9]+/)[0] != AlchemyFlux::VERSION
+    raise "Latest version in CHANGELOG.md does not match AlchemyFlux::VERSION"
+  end
```
Comment (pond) [inline Rakefile:25]: That's a really nice touch 👍

**Pull request merged**